### PR TITLE
CHANGELOG: add 0.3.10 release

### PR DESCRIPTION
**Pull request by terrafrost**
See #496 . I figure this should be merged into the 2.0 branch and the master branch as well. Should 2.0.0 be mentioned in the changelog? Since 2.0.0 isn't even tagged it's not entirely clear to me if that version has even been officially released. Regardless, if it has, that can be another pull request.

### CODE REVIEW

**Comment by bantu** [comment]
You should be requesting this to be merged into 1.0 then, right?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## 0.3.10 - 2015-02-04
+
+- simplify window size handling ([#538](https://github.com/phpseclib/phpseclib/pull/538))
```
Comment (bantu) [inline CHANGELOG.md]: should probably supply some context here. ssh?

**Comment by bantu** [comment]
Not sure about the best way to handle changelog. Symfony uses version specific changelogs. E.g. CHANGELOG-1.0.md and CHANGELOG-2.0.md.

**Comment by GrahamCampbell** [comment]
Why don't we have a 0.3 branch? Surely we should have one, even if it's not maintained?

**Comment by bantu** [comment]
@GrahamCampbell 0.3.x becomes 1.0.0 when OpenSSL support is implemented. So 0.3 is in the 1.0 branch and fully BC.

**Comment by GrahamCampbell** [comment]
Right, so you're just pushing out a final 0.3 release from the 1.0 branch then. Makes sense. I've done something similar before. :)

**Comment by bantu** [comment]
@GrahamCampbell My understand was that terrafrost wants to maintain 1.0 for a longer time. Possibly lower than 2.0. Kind of as an LTS release.

**Comment by terrafrost** [comment]
> You should be requesting this to be merged into 1.0 then, right?

Yup. idk how this should be handled for the 2.0 series.

> @GrahamCampbell My understanding is that terrafrost wants to maintain 1.0 for a longer time. Possibly longer than 2.0. Kind of as an LTS release.

Yah - pretty much.

File: CHANGELOG.md (hunk 1 above)
Comment (terrafrost) [inline CHANGELOG.md]: Yah - it's for SSH. SSH2 specifically. I'll update it later today.

**Comment by bantu** [comment]
> Yup. idk how this should be handled for the 2.0 series.

Rename CHANGELOG.md to CHANGELOG-1.0.md in 1.0 branch. Merge down. Add CHANGELOG-2.0 to 2.0 branch. Merge down. ?

**Comment by GrahamCampbell** [comment]
> Rename CHANGELOG.md to CHANGELOG-1.0.md in 1.0 branch. Merge down. Add CHANGELOG-2.0 to 2.0 branch. Merge down. ?

:+1: :rocket:

**Comment by ravage84** [comment]
http://keepachangelog.com/ has some ideas about handling change logs.

I personally would keep only one change log.
Starting a new one for each new major release (SemVer-wise).
All pre-stable releases are also kept in this change log.
So all 0.x.x releases including all beta and RC releases of 1.x would be kept in the change log which is included in the code tree of 1.x

**Comment by terrafrost** [comment]
@ravage84 - you're proposing that each branch have it's own changelog independent of any other branch? eg.

CHANGELOG.md for the 2.0 branch starts at 2.0, as tho the 0.x/1.x series never happened?

@AdamWill - what are your thoughts?

**Comment by AdamWill** [comment]
Not sure why you're asking me, for a distro perspective? Don't think I have a strong opinion or Fedora / distros in general have particular policies, it's not at all uncommon to have apps which don't have all their history available in current tarballs for sure. So long as it's all available *somewhere* I don't think I really have an opinion.

**Comment by ravage84** [comment]
@terrafrost
The CHANGELOG.md for the 2.0 branch starts with whatever version was developed for this branch first (normally a pre stable release, such as 2.0.0-beta1). So people can see the changes that happened towards the first major stable of that branch.
I would not mind having the full back log of changes that happened before that branch, but it creates an interesting problem when you have more than one branch actively maintained.
Also I don't think there is much of a benefit of having the full change log in a branch of  a later release, because if you are using the 2.x release and really need to know what changed within the lifecycle of the 1.x release you can simple look it up in that branch.

**Comment by cnelissen** [comment]
I concur with @ravage84... Each branch really only needs it's own history. If you are interested in history from a different branch, then you could go look at it.

**Comment by bantu** [comment]
A single changelog per branch likely results in merge conflicts when merging e.g. 1.0 into 2.0. This is not a big problem, but should be considered.

**Comment by bantu** [comment]
Replaced by #624

**Pull request closed without merging**